lib/cubit/ln_address/services/webhook_request_builder.dart

```diff
@@ -28,17 +28,20 @@ class WebhookRequestBuilder {
   Future<RegisterLnurlPayRequest> buildRegisterRequest({
     required String webhookUrl,
     String? username,
+    String? offer,
   }) async {
     try {
       _logger.info('Building registration request for webhook: $webhookUrl');
 
       // Format the username component if provided
-      final String usernameComponent = _formatUsernameComponent(username);
+      final String usernameComponent = _formatOptionalComponent(username);
+      final String offerComponent = _formatOptionalComponent(offer);
+      final String additionalData = '$usernameComponent$offerComponent';
```

Comment:
Let's extract these into a `_formatAdditionalData` fn or, perhaps inlining it like:
```
final additionalData = '${_formatOptionalComponent(username)}${_formatOptionalComponent(offer)}';
```

`_formatOptionalComponent` could also take a list of arguments.